autocomplete-plugin-algolia-insights can not be installed alongside search-insights@2.x

## Description

Even though [search-insights@2](https://github.com/algolia/search-insights.js/releases/tag/v2.0.0) does not seem to contain any breaking changes that are truly incompatible with `autocomplete-plugin-algolia-insights`, npm refuses to install the latest versions of both the same time.

## Reproduction

```
$ npm --version
8.3.1
$ npm i @algolia/autocomplete-plugin-algolia-insights search-insights@2
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: undefined@undefined
npm ERR! Found: search-insights@2.2.1
npm ERR! node_modules/search-insights
npm ERR!   search-insights@"2" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer search-insights@"^1.0.0" from @algolia/autocomplete-plugin-algolia-insights@1.5.6
npm ERR! node_modules/@algolia/autocomplete-plugin-algolia-insights
npm ERR!   @algolia/autocomplete-plugin-algolia-insights@"*" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

## Solution

Merge and release https://github.com/algolia/autocomplete/pull/717

--- a/packages/autocomplete-plugin-algolia-insights/package.json
+++ b/packages/autocomplete-plugin-algolia-insights/package.json
@@ -38,6 +38,6 @@
     "@algolia/autocomplete-core": "1.5.6"
   },
   "peerDependencies": {
-    "search-insights": "^1.0.0"
+    "search-insights": ">= 1 < 3"
   }
 }

